docs/tutorial/graphical.rst

```diff
@@ -458,3 +458,138 @@ Note that in this example, :code:`param_A` and :code:`param_B` are identically g
+           for i in range(self.iterations):
+               data = {
+                   'Iteration': i,
+                   'Random Number 1': random.random(),
+                   'Random Number 2': random.random(),
+                   'Random Number 3': random.random()
+               }
+               self.emit('results', data)
+               log.debug("Emitting results: %s" % data)
+               self.emit('progress', 100 * i / self.iterations)
+               sleep(self.delay)
+               if self.should_stop():
+                   log.warning("Caught the stop flag in the procedure")
+                   break
+
+
+   class MainWindow(ManagedDockWindow):
+
+       def __init__(self):
+           super().__init__(
+               procedure_class=RandomProcedure,
+               inputs=['iterations', 'delay', 'seed'],
+               displays=['iterations', 'delay', 'seed'],
+               x_axis=['Iteration', 'Random Number 1'],
+               y_axis=['Random Number 1','Random Number 2', 'Random Number 3']
+           )
+           self.setWindowTitle('GUI Example')
+
+       def queue(self):
+           filename = tempfile.mktemp()
+
+           procedure = self.make_procedure()
+           results = Results(procedure, filename)
+           experiment = self.new_experiment(results)
+
+           self.manager.queue(experiment)
+
+
+   if __name__ == "__main__":
+       app = QtWidgets.QApplication(sys.argv)
+       window = MainWindow()
+       window.show()
+       sys.exit(app.exec())
+
+Now we can see our :code:`ManagedDockWindow`:
+
+.. image:: managed_dock_window.png
+    :alt: Managed dock window
+
+As you can see from the above screenshot, our example code created three docks with following "X Axis" and "Y Axis" labels:
+
+1. **X Axis:** "Iteration"        **Y Axis:** "Random Number 1"
+2. **X Axis:** "Random Number 1"  **Y Axis:** "Random Number 2"
+3. **X Axis:** "Random Number 1"  **Y Axis:** "Random Number 3"
+
+The list of strings for :code:`x_axis` and :code:`y_axis` set the default labels for each dockable plot and the longest list determines how many dockable plots are created. To highlight this point, in our example we define :code:`x_axis` and :code:`y_axis` with the following lists::
+
+   x_axis=['Iteration', 'Random Number 1'],
+   y_axis=['Random Number 1','Random Number 2', 'Random Number 3']
+
+If one list is longer than the last element if the other list is used as the default label for the rest of the dockable plots.
+In our example that is why we have two **X Axis** labels with "Random Number 1".
+The longest list between :code:`x_axis` and :code:`y_axis` determines the number of plots.
+In our example :code:`y_axis` has the longest list with a length of three so three plots are created.
+
+You can pop out a dockable plot from the main dock window to its own window by double clicking the blue "Dock #" title bar, which is to the left of each plot by default:
```

Comment:
Oh, this is really cool

Reply:
Yea the pyqtgraph library really is awesome